hypervisor/arch/x86/pagetable.c

```diff
@@ -455,13 +569,53 @@ void pgtable_add_map(uint64_t *pml4_page, uint64_t paddr_base, uint64_t vaddr_ba
 	}
 }
 
+/**
+ * @brief Create a new root page table.
+ *
+ * This function initializes and returns a new root page table. It is typically used during the setup of a new execution
+ * context, such as initializing a hypervisor PML4 table or creating a virtual machine. The root page table is essential
+ * for defining the virtual memory layout for the context.
+ *
+ * It creats and initializes a root page table, then returns the pointer.
+ *
+ * @param[in] table Pointer to the struct pgtable used by the new root page table.
+ *
+ * @return A pointer to the newly created root page table.
+ *
+ * @pre table != NULL
+ *
+ * @post N/A
+ */
 void *pgtable_create_root(const struct pgtable *table)
 {
 	uint64_t *page = (uint64_t *)alloc_page(table->pool);
 	sanitize_pte(page, table);
 	return page;
 }
 
+/**
+ * @brief Create a root page table for Secure World.
+ *
+ * This function initializes a new root page table for Secure World. It is intended to be used during the initialization
+ * phase of Trusty, setting up isolated memory regions for secure execution. Secure world can access Normal World's
+ * memory, but Normal World cannot access Secure World's memory. The PML4T/PDPT for Secure World are separated from
+ * Normal World. PDT/PT are shared in both Secure World's EPT and Normal World's EPT. So this function copies the PDPTEs
+ * from the Normal World to the Secure World.
+ *
+ * It creats and initializes a root page table for Secure World, and then returns the pointer.
+ *
+ * @param[inout] table Pointer to the struct pgtable used by the new root page table.
```

Comment:
1. `inout` -> `in`
2. Suggest: `A pointer to the struct pgtable which provides the page pool and callback functions to be used when creating the new page.`